### PR TITLE
Add User classes and changes name of Enum folder User

### PR DESCRIPTION
**Pull request by mooneity**
Adds the classes User, NewUser, Patient, Staff, and Admin into the backend folder. The Enum folder User is also changed to Users in order to avoid collisions between said folder and User.java and User class.

### CODE REVIEW

File: backend/User.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    } // end setRoleID()
+
+    public void setPassword(String password) {
+        this.password = password;
+    } // end setPassword()
+}
+
+class Patient extends User {
+    // Attributes
+    private int roleID = 0;
```
Comment (scritty1249) [inline backend/User.java]: can be final

File: backend/User.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    // Removes appointment from upcomingAppointments
+    public void removeAppointment(Appointment appointment){
+       this.upcomingAppointment.remove(appointment);
+    }
+    */
+ }
+
+ class Staff extends User {
+    // Private Attributes
+    private int roleID = 1;
```
Comment (scritty1249) [inline backend/User.java]: can be final

File: backend/User.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    
+    // Setter for hourlyRate
+    public void setHourlyRate(double hourlyRate){
+      this.hourlyRate = hourlyRate;
+    }
+}
+
+class Admin extends Staff {
+    // attributes
+    private int roleID = 2;
```
Comment (scritty1249) [inline backend/User.java]: can be final

File: backend/User.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    @Override
+    // get the roleID
+    public int getRoleID(){
+      return roleID;
+    }
+
+    @Override
+    // get the hourly rate
+    public double getHourlyRate() {
+        return super.getHourlyRate();
```
Comment (scritty1249) [inline backend/User.java]: if we're extending the class, it should inherit these methods. Why call super?

File: backend/User.java [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    } // end getHourlyRate()
+
+    @Override
+    public void setHourlyRate(double hourlyRate) {
+        super.setHourlyRate(hourlyRate);
```
Comment (scritty1249) [inline backend/User.java]: if we're extending the class, it should inherit these methods. Why call super?

File: backend/User.java [hunk 6]
```diff
@@ -0,0 +1,226 @@
+package backend;
```
Comment (scritty1249) [inline backend/User.java]: No such thing as a "frontend" in the context of running our Java code. Consequently, I don't think there's a need for an explicit "backend" package to exist.

**Review by scritty1249 — CHANGES_REQUESTED** [review]
See comments

File: backend/User.java (hunk 6 above)
Comment (W-Sammy) [reply backend/User.java]: If no such package is necessary, please remove from final commit before approved PR

**Review by scritty1249 — DISMISSED** [review]

**Comment by scritty1249** [comment]
LGTM, thanks for your work!

File: backend/Users/Enum/RoleConstant.java [hunk 7]
```diff
@@ -1,5 +1,3 @@
-package backend.Users.Enum;
```
Comment (scritty1249) [inline backend/Users/Enum/RoleConstant.java]: package Users.Enum;

File: backend/User.java (hunk 6 above)
Comment (scritty1249) [reply backend/User.java]: add a commit removing this?

File: backend/Users/Enum/RoleConstant.java [hunk 8]
```diff
@@ -1,5 +1,3 @@
-package backend.User.Enum;
```
Comment (scritty1249) [inline backend/Users/Enum/RoleConstant.java:1]: change to `package Users.Enum;`

**Review by erdsmabilog — COMMENTED** [review]
For user class and all subsequent uses of these attributes -- how many bytes can these hold in java?
Sorry I think I may need to update database schema, but these will both hold a hash -- 256 bits:
    // attributes
    private byte userID;
    private int roleID = 0;

**Comment by mooneity** [comment]
Based on the UML Diagram in our Software Design Document, I assume the maximum is 32.

File: backend/User.java [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+    public void setBirthDate(int birthDate) {
+        this.birthDate = birthDate;
+    } // end setBirthDate()
+}
+
+class NewUser extends User {
+    // attributes
+    private String password;
+
```
Comment (W-Sammy) [inline backend/User.java:107]: Is userID supposed to be generated on account creation? Or where are we pulling that information from?

File: backend/User.java [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+    // Adds an appointment to upcomingAppointments
+    public void addAppointment(Appointment appointment){
+       this.upcomingAppointments.add(appointment);
+    }
+    
+    // Removes appointment from upcomingAppointments
+    public void removeAppointment(Appointment appointment){
+       this.upcomingAppointment.remove(appointment);
+    }
+    */
```
Comment (W-Sammy) [inline backend/User.java:153]: Commented out code intentional?

File: backend/User.java [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+        super(userID, roleID, firstName, lastName, address, sex, phoneNumber, email, birthDate, hourlyRate);
+    } // end constructor
+
+    @Override
+    // get the roleID
+    public int getRoleID(){
+      return roleID;
+    }
+
+    @Override
```
Comment (W-Sammy) [inline backend/User.java]: Is an override for hourlyRate required if Admin extends Staff? When Admins are created, their pay should be instantiated in the staff class.

**Review by W-Sammy — COMMENTED** [review]
Couple comments addressing code clarity and possible concerns

File: backend/User.java (hunk 10 above)
Comment (W-Sammy) [reply backend/User.java:153]: Is this commented out portion of code done intentially?*

Comment (mooneity) [reply backend/User.java:153]: This portion of the code has been commented out for later use, so yes.

File: backend/User.java (hunk 9 above)
Comment (scritty1249) [reply backend/User.java:107]: Yes, generated on account creation.

**Review by scritty1249 — APPROVED** [review]
LGTM

**Review by cscjohn — APPROVED** [review]

**Comment by mooneity** [comment]
Ignore the commit Testing as there are no changes and there were simply issues from my side.

**Pull request merged**